### PR TITLE
Streams: update tests for Web IDL conversion

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/streams/pull/1035. Notable changes:

- Updates for the normative changes listed there.
- Introduce an idlharness test
- Remove various tests for things that are covered by idlharness, such as brand checks, the order of getting values from dictionaries, etc.
- Updated for the fact that everything is now globally exposed, so some roundabout code to get constructors can be removed.
- Slight timing updates: the pattern of returning a promise from an underlying sink/source/transformer `start()` function, then waiting on that before doing asserts, does not work with Web IDL's "promise resolved with". So instead we use `flushAsyncEvents()` to wait a little longer.
- Consolidated queuing strategy tests into a single file, since after deleting the things covered by idlharness, most of the tests are shared.
- Added tests that underlyingSink/underlyingSource are converted after queuingStrategy, since those dictionary conversions are done in prose.
- Added a test for various updates to the Web IDL async iterators specification

### CODE REVIEW

**Review by ricea — APPROVED** [review]
lgtm

The removed tests will be done by idlharness.js, so we definitely don't want to keep them.

As you say, it would be good to verify that the second argument is unpacked before the second, but we don't need all this infrastructure to do it, just something like
```javascript
assert_throws_exact(error2, 
    () => new ReadableStream({ get start() { throw error1; } }, { get size() { throw error2; } }));
```
should do it.

File: streams/queuing-strategies.any.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      'construction fails with an object with a throwing highWaterMark getter');
+
+    assert_equals((new QueuingStrategy(highWaterMarkObjectGetter)).highWaterMark, highWaterMark);
+  }, `${QueuingStrategy.name}: Constructor behaves as expected with strange arguments`);
+
+  test(() => {
+    for (const [input, output] of highWaterMarkConversions.entries()) {
+      const strategy = new QueuingStrategy({ highWaterMark: input });
+      assert_equals(strategy.highWaterMark, output, `${input} gets set correctly`);
+   }
```
Comment (ricea) [inline streams/queuing-strategies.any.js:47]: Indentation looks wrong.

File: streams/readable-byte-streams/general.any.js [hunk 2]
```diff
@@ -337,8 +337,8 @@ promise_test(() => {
   return Promise.resolve().then(() => {
     assert_equals(pullCount, 1, 'pull() must have been invoked once');
     const byobRequest = byobRequests[0];
-    assert_true(byobRequest.defined, 'first byobRequest must not be undefined');
-    assert_true(byobRequest.viewDefined, 'first byobRequest.view must not be undefined');
+    assert_true(byobRequest.nonNull, 'first byobRequest must not be null');
+    assert_true(byobRequest.viewDefined, 'first byobRequest.view must not be null');
```
Comment (MattiasBuelens) [inline streams/readable-byte-streams/general.any.js:341]: `byobRequest.viewDefined` is set to `view !== undefined`, so the message should be "first byobRequest.view must not be undefined" (as it was before).

Comment (domenic) [reply streams/readable-byte-streams/general.any.js:341]: Seems like this also needs to become nullable.

**Comment by ricea** [comment]
Everything still lgtm. I don't know what to do about ReadableStreamBYOBRequest. Could we just call idl_test() asynchronously after grabbing one?

**Comment by LukeZielinski** [comment]
Looks like TaskCluster didn't run here, retrying.

**Comment by stephenmcgruer** [comment]
> Looks like TaskCluster didn't run here, retrying.

This is a draft PR, TaskCluster doesn't run on drafts.

**Comment by foolip** [comment]
Taskcluster does run on draft PRs after https://github.com/web-platform-tests/wpt/pull/23566, but that also dropped the "ready_for_review" trigger. That's why Taskcluster still hasn't run here. I'll close and reopen as a workaround. (Rebasing would also work.)

**Review by ricea — APPROVED** [review]
I re-reviewed everything. Still lgtm.

**Comment by ricea** [comment]
Updates lgtm.

File: streams/readable-streams/async-iterator.any.js [hunk 3]
```diff
@@ -292,20 +292,256 @@ promise_test(async t => {
...
+  const iterResult1 = await it.next();
+  assert_equals(iterResult1.value, 0, '1st next() value');
+  assert_equals(iterResult1.done, false, '1st next() done');
+
+  await promise_rejects_exactly(t, error1, it.next(), '2nd next()');
+
+  const iterResult3 = await it.return('return value');
+  assert_equals(iterResult3.value, 'return value', 'return() value');
+  assert_equals(iterResult3.done, true, 'return() done');
+}, 'return() after next() reports an error');
```
Comment (TimothyGu) [inline streams/readable-streams/async-iterator.any.js:415]: nit: the English phrase has a bit of an ambiguity. I first read it as "[return() after next()] reports an error", only afterwards did I realize it's actually "return() after [next() reports an error]". This also applies to the "next() after next()" tests above.

**Review by TimothyGu — APPROVED** [review]
The recent additions around `next`/`return` interactions look good, modulo one nit. Also, would it be a good idea to have an `assert_iteration_result()`, perhaps for future?

**Pull request merged**